packages/typegpu/src/tgsl/wgslGenerator.ts

```diff
@@ -192,6 +192,11 @@ const opCodeToCodegen = {
   '-': sub[$internal].gpuImpl,
   '*': mul[$internal].gpuImpl,
   '/': div[$internal].gpuImpl,
+  '**': (lhsExpr: Snippet, rhsExpr: Snippet) => {
+    const [convLhs, convRhs] =
+      convertToCommonType([lhsExpr, rhsExpr], [f32, f16]) ?? [lhsExpr, rhsExpr];
+    return snip(stitch`pow(${convLhs}, ${convRhs})`, convLhs.dataType);
+  },
```

Comment:
We could move `pow` to the new `dualImpl` api like:
```ts
const powCpu = <T extends AnyFloatVecInstance | number>(
  base: T,
  exponent: T,
): T => {
  if (typeof base === 'number' && typeof exponent === 'number') {
    return (base ** exponent) as T;
  }
  if (
    typeof base === 'object' &&
    typeof exponent === 'object' &&
    'kind' in base &&
    'kind' in exponent
  ) {
    return VectorOps.pow[base.kind](base, exponent) as T;
  }
  throw new Error('Invalid arguments to pow()');
};

export const pow: PowOverload = dualImpl({
  name: 'pow',
  signature: (...args) => {
    const uargs = unify(args, [f32, f16, abstractFloat]) ?? args;
    return {
      argTypes: uargs,
      returnType: isNumericSchema(uargs[0]) ? uargs[1] : uargs[0],
    };
  },
  normalImpl: powCpu,
  codegenImpl: (lhs, rhs) => stitch`pow(${lhs}, ${rhs})`,
});
```
Then this simplifies to:
```suggestion
  '**': pow[$internal].gpuImpl,
```
And we get some free comptime powing

Reply:
```diff
  "fn power() {
-   var n = pow(2, 4);
+   var n = 16.;
  }"
```